cms/djangoapps/contentstore/views/item.py

```diff
@@ -194,13 +201,41 @@ def xblock_handler(request, usage_key_string):
                 request.json.get('display_name'),
             )
 
-            return JsonResponse({"locator": unicode(dest_usage_key), "courseKey": unicode(dest_usage_key.course_key)})
+            return JsonResponse({'locator': unicode(dest_usage_key), 'courseKey': unicode(dest_usage_key.course_key)})
+        elif 'move_source_locator' in request.json:
+            move_source_usage_key = usage_key_with_run(request.json['move_source_locator'])
+            dest_parent_usage_key = usage_key_with_run(request.json['dest_source_locator'])
+            source_index = request.json.get('source_index', None)
+
+            source_course = move_source_usage_key.course_key
+            dest_course = dest_parent_usage_key.course_key
+            if (
+                    not has_studio_write_access(request.user, dest_course) or
+                    not has_studio_read_access(request.user, source_course)
+            ):
+                raise PermissionDenied()
+
+            is_action_valid = is_valid_move(move_source_usage_key, dest_parent_usage_key)
+            if not is_action_valid:
+                error = _('You can not move {source_type} into {dest_type}').format(
+                    source_type=move_source_usage_key.block_type,
+                    dest_type=dest_parent_usage_key.block_type,
+                )
+                return JsonResponse({'error': error}, status=400)
+            actual_source_index = _get_source_index(move_source_usage_key)
+            _move_item(move_source_usage_key, dest_parent_usage_key, request.user, source_index)
+            context = {
+                'move_source_locator': unicode(move_source_usage_key),
+                'dest_source_locator': unicode(dest_parent_usage_key),
+                'source_index': source_index if source_index is not None else actual_source_index
```

Comment:
Why do you want to return the source index here? I can't think of a useful situation to know what the old index was. IMO this should just be ``index`` and it should be for the new location.